Library/Homebrew/livecheck/strategy.rb

```diff
@@ -80,6 +80,47 @@ def from_url(url, livecheck_strategy: nil, regex_provided: nil)
           (strategy.const_defined?(:PRIORITY) ? -strategy::PRIORITY : -DEFAULT_PRIORITY)
         end
       end
+
+      # Fetches the content at the URL and returns a hash containing the
+      # content and, if there are any redirections, the final URL.
+      # If `curl` encounters an error, the hash will contain a `:messages`
+      # array with the error message instead.
+      # @param url [String] the URL of the content to check
+      # @param user_agent [Symbol] the symbol for a corresponding user agent
+      # @return [Hash]
+      def self.page_content(url, user_agent: :default)
+        original_url = url
+
+        args = curl_args(
+          "--compressed", # Request/handle compressed responses
+          "--include",    # Include HTTP response headers in the output
+          "--location",   # Follow redirects
+          user_agent: user_agent,
+        )
+        result = curl_with_workarounds(
+          *args, url,
+          print_stdout: false, print_stderr: false,
+          debug: false, verbose: false
+        )
+
+        if result.status
```

Comment:
Do you mean 

```suggestion
        unless result.status.success?
```

?

Reply:
Yes, that's certainly a better representation of what I was aiming to achieve and I've confirmed that we still identify errors as expected. I'm not sure what I was thinking when I originally wrote that but I've been staying up too late working on this recently, ahah.

This change is included in the latest push.